### ISSUE TITLE
Inner declaration missing for initialization graphs

### ISSUE THREAD

**Issue by Jand42** [fsharp/FSharp.Compiler.Service#497]
(This is quite possibly related to https://github.com/fsharp/FSharp.Compiler.Service/issues/495 if the problem is around module-level function values.)

This small snippet:

``` fsharp
let rec badLoop =
    () // so that it is a function value
    fun x -> badLoop (x + 1)           
```

is reported by FCS as `let badLoop = badLoop@4.Force(())`, but the declaration for `badLoop@4` is not present inside the `FSharpImplementationFileContents.Declarations` list.

**Comment by dsyme** [comment]
Nice find :)

**Comment by dsyme** [comment]
I checked this and it seems the item is present in the Declarations - at least it is now.  It is marked CompilerGenerated so you have to be careful that you are not skipping those

**Comment by Jand42** [comment]
@dsyme Thanks for the update.

Tested it now (at commit 5a87eb9), still the exposed members are not usable enough for translation:

* On generated members, both `EnclosingEntity` and `LogicalEnclosingEntity` are throwing an exception, making it ambigous which module contains them.
* The recursive reference is exposed as a `Value x` pattern, not a `Call`, where `x.IsModuleValueOrMember` returns false, so it seems like a local variable where it is not.

**Comment by dsyme** [comment]
@Jand42  Ah ok, thanks
